staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -223,13 +228,9 @@ func (flags *ApplyFlags) AddFlags(cmd *cobra.Command) {
 	cmdutil.AddServerSideApplyFlags(cmd)
 	cmdutil.AddFieldManagerFlagVar(cmd, &flags.FieldManager, FieldManagerClientSideApply)
 	cmdutil.AddLabelSelectorFlagVar(cmd, &flags.Selector)
+	cmdutil.AddPruningFlags(cmd, &flags.Prune, &flags.PruneAllowlist, &flags.PruneWhitelist, &flags.All, &flags.ApplySetRef)
 
 	cmd.Flags().BoolVar(&flags.Overwrite, "overwrite", flags.Overwrite, "Automatically resolve conflicts between the modified and live configuration by using values from the modified configuration")
-	cmd.Flags().BoolVar(&flags.Prune, "prune", flags.Prune, "Automatically delete resource objects, that do not appear in the configs and are created by either apply or create --save-config. Should be used with either -l or --all.")
-	cmd.Flags().BoolVar(&flags.All, "all", flags.All, "Select all resources in the namespace of the specified resource types.")
```

Comment:
Nit: I wouldn't consider `--all` a prune-only flag, so I'd suggest to leave it here.

Reply:
I know that `--selector` affects the main `apply` operation, but I thought that `--all` only affects pruning. I found that it was definitely added for pruning: https://github.com/kubernetes/kubernetes/commit/56a22f925f6f1fd774ad1ae9e04bcf8d75bbde63#diff-e255643fe2dc451729fc438dad24902dff489a8801a5c976c2d783649ceb870bR103. What else does it do now? Based on the code, it is only used during flag validation, to ensure the user intended to target everything if they omitted `--selector`.

That PR also shows that `--grace-period` and `--cascade` were also added for pruning originally, but based on the coded I believe that those also affect the `--force` flag these days.